app/services/save_auction.rb

```diff
@@ -4,10 +4,23 @@ def initialize(auction)
   end
 
   def perform
-    auction.save
+    saved = auction.save
+    schedule_auction_ended_job(saved)
+
+    saved
   end
 
   private
 
   attr_reader :auction
+
+  def schedule_auction_ended_job(saved)
+    if should_schedule_auction_ended_job?(saved)
+      AuctionEnded.new(auction).delay(run_at: auction.ended_at).perform
+    end
+  end
+
+  def should_schedule_auction_ended_job?(saved)
+    saved && !auction.ended_at.nil?
```

Comment:
Is it worth removing the `if` clause? Still might be a nice check to have in place.